storage/datastore.go

```diff
@@ -76,6 +76,17 @@ func (ds *DataStore) Read(txn Transaction, path ast.Ref) (interface{}, error) {
 	return ds.GetRef(path)
 }
 
+// Write modifies a document referred to by path.
+func (ds *DataStore) Write(txn Transaction, op PatchOp, path ast.Ref, value interface{}) error {
+	p, err := path.Underlying()
+	if err != nil {
+		return err
+	}
+	// TODO(tsandall): Patch() assumes that paths in writes are relative to
+	// "data" so drop the head here.
+	return ds.Patch(op, p[1:], value)
+}
```

Comment:
This seems cool.  When is OPA writing to the backend?  

We may want to make it super easy for the user to ensure OPA never does a write (for a particular datastore).  Maybe even make writing disabled by default.